### PR TITLE
Allow excon 1.x and require base64 for modern rubies

### PR DESCRIPTION
**Pull request by zarqman**
This PR updates the gemspec to:
* Allow `excon` 1.x
* Update `excon` minimum to 0.97, which appears to have already been required anyway
* Require `base64` to silence the deprecation warning on Ruby 3.3

Local tests have been flaky and the swarm tests don't seem to run with my dockerd at all. However, apart from the swarm tests, I was able to get everything else to pass once. That _might_ require updates to some of the GitHub URLs. I've excluded those updates from this PR as I'm not sure if they are really required or not. I'm happy to add them to the PR if that'd be helpful.

### CODE REVIEW

**Comment by nu12** [comment]
Hi @zarqman , thank you for this PR, your contribution is appreciated.

I believe some responses changed from 1.43 to 1.46, that would require some refactoring of the tests so that they pass. We can either add it to this PR or merge another branch before merging this one. In any case, we might also bump bundler version as the pipeline is not working at all.

Any thoughts on bumping min ruby version to 3.0  instead of 2.3 ?

**Comment by nu12** [comment]
Let's group all versioning modification in this PR. If you can manage to fix the tests too, it will be welcomed.
I acknowledge that tests have always been inconsistent. Maybe this is something to be improved later (in the form of simplification).

For the API version, this is what we have in the documentation:

> A given version of the Docker Engine SDK supports a specific version of the Docker Engine API, as well as all earlier versions.

> The Docker API is backward-compatible, so you don't need to update code that uses the API unless you need to take advantage of new features.

And, more important: 

> When using curl directly, specify the version as the first part of the URL. For instance, if the endpoint is /containers/ you can use /v1.46/containers/

Source: https://docs.docker.com/reference/api/engine/

But I'd like to implement this part in a separate PR.

**Comment by nu12** [comment]
Hi @zarqman  

Support to older API versions is implemented with #9.

I had to make a few adjustments to pass the tests, everything should be good to go now. 

You can rebase your branch to integrate these changes. Is it possible to upgrade minimum excon version to 1.x instead of 0.97? 

I'll wait for this PR to release a new version. Thank you.

**Comment by zarqman** [comment]
Rebased and Ruby bumped to 3.0. I took the liberty to cleanup some Ruby warnings too.

Between #9 and a couple other minor changes, tests run and pass on both excon 0.97 and 1.1.1. 

Since excon was at 0.x for so long, and some other gems still have it locked to 0.x, I left the excon minimum at 0.97. dockerapi's usage of excon is pretty basic and both 0.x and 1.x work fine for now.

Let me know if anything else needs to be changed.

**Comment by nu12** [comment]
Just one thing, encapsulating print_response_to_stdout gives an error if we call the getter without initializing the variable:
image

**Comment by zarqman** [comment]
Good catch. Fixed!

**Review by nu12 — APPROVED** [review]
LGTM

**Comment by nu12** [comment]
Thank you @zarqman

**Pull request merged**
